src/App.tsx

```diff
@@ -1,26 +1,272 @@
+        setIsLoading(false);
+      });
+  }, [todos]);
+
+  const filteredTodos = useMemo(() => {
+    switch (status) {
+      case TodoStatus.Active:
+        return todos.filter(todo => !todo.completed);
+      case TodoStatus.Completed:
+        return todos.filter(todo => todo.completed);
+      default:
+        return todos;
+    }
+  }, [status, todos]);
+
+  const hasCompletedTodos = useMemo(() => {
+    return todos.some(todo => todo.completed);
+  }, [todos]);
+
+  const handleDeleteTodo = (deletedTodoId: number) => {
+    setTodos(prevTodos => prevTodos.filter(todo => todo.id !== deletedTodoId));
+  };
+
+  const handleDeleteCompleted = () => {
+    todos.forEach(todo => {
+      if (!todo.completed) {
+        return;
+      }
+
+      Services.deleteTodo(todo.id)
+        .then(() => {
+          setTodos(prevTodo =>
+            prevTodo.filter(currTodo => currTodo.id !== todo.id),
+          );
+        })
+        .catch(() => {
+          setErrorTitle(ErrorTypes.UnableToDeleteTodo);
+        });
+    });
+  };
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">
-          React Todo App - Load Todos
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          {todos.length > 0 && (
+            <button
+              onClick={handleToggleAll}
+              type="button"
+              className={`todoapp__toggle-all ${todos.every(todo => todo.completed) ? 'active' : ''}`}
+              data-cy="ToggleAllButton"
+            />
+          )}
+
+          <form onSubmit={handleAddTodo}>
+            <input
+              data-cy="NewTodoField"
+              type="text"
+              className="todoapp__new-todo"
+              placeholder="What needs to be done?"
+              ref={inputRef}
+              value={newTodoTitle}
+              onChange={e => setNewTodoTitle(e.target.value)}
+              disabled={isLoading}
+            />
+          </form>
+        </header>
+        <section className="todoapp__main" data-cy="TodoList">
+          {filteredTodos.map(todo => (
+            <TodoComponent
+              key={todo.id}
+              todo={todo}
+              onTodoChange={handleTodoChange}
+              onDeleteTodo={handleDeleteTodo}
+              onError={errorMessage => setErrorTitle(errorMessage)}
+            />
+          ))}
+          {tempTodo && (
+            <TodoComponent
+              todo={tempTodo}
+              isTemp={true}
+              onTodoChange={handleTodoChange}
+              onError={errorMessage => setErrorTitle(errorMessage)}
```

Comment:
this onError function is being repeated, you may add a handler to simplify it - even though it's a simple function, using a handle helps you debug and improves overall code readability